### PR TITLE
Add x509 parsing utilities

### PR DESCRIPTION
**Pull request by rgnote**
* Moving x509 cert, key parsing utilities from notation-go to notation-core-go
* Will raise a separate PR for refactoring notation-go and notation repos

Signed-off-by: rgnote <5878554+rgnote@users.noreply.github.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/notaryproject/notation-core-go
+
+go 1.18
```
Comment (rgnote) [inline go.mod:3]: I want someone with more familiarity with golang chime in here. Currently, notation and notation-go are on 1.17. Reading 1.18 [release notes](https://go.dev/doc/go1.18), it looks like 1.18 is backward compatible and it is also fixing some bugs present in earlier Go versions. Let me know if there a reason we shouldn't upgrade to 1.18.

Comment (gokarnm) [reply go.mod:3]: @shizhMSFT , @qmuntal, any feedback here?

Comment (gokarnm) [reply go.mod:3]: @shizhMSFT , @qmuntal, any feedback here?

Comment (qmuntal) [reply go.mod:3]: The `go` directive inside `go.mod` does not refer to the go version that will be used to compile the library, but to the maximum set of language features that this library supports. For example, generics are only available if we set it to `1.18` and `embed` if we set it to `1.16` or greater.

A customer using a lower Go toolchain version can still use a module whose `go` directive if set to a higher version unless the module really uses a newer language feature.

The best practice is to set the `go` directive to the lowest supported Go version. As we are not using generics here, I would set it to `1.17` to be in-line with `notation-go`.

Comment (rgnote) [reply go.mod]: Good to know. Updated the PR.

@gokarnm Can you merge if this looks good?

File: x509/cert.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+func ReadCertificateFile(path string) ([]*x509.Certificate, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	return ParseCertificatePEM(data)
+}
+
+// ParseCertificatePEM parses a certificate PEM.
+func ParseCertificatePEM(data []byte) ([]*x509.Certificate, error) {
```
Comment (gokarnm) [inline x509/cert.go:19]:
```suggestion
// ParseCertificatePEM parses a certificate PEM,
// Does not perform any additional validations if the certs are a valid cert chain.
func ParseCertificatePEM(data []byte) ([]*x509.Certificate, error) {
```

**Review by gokarnm — APPROVED** [review]

**Comment by dtzar** [comment]
LGTM

**Review by jondonas — APPROVED** [review]

**Review by qmuntal — APPROVED** [review]

**Review by dtzar — APPROVED** [review]

**Review by NiazFK — APPROVED** [review]
LGTM

**Comment by NiazFK** [comment]
Add x509 parsing utilities

Signed-off-by: rgnote <5878554+rgnote@users.noreply.github.com>
Co-authored-by: Milind Gokarn <milind81@gmail.com>

**Pull request merged**